core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -259,16 +274,17 @@ protected OAuth2Util.AuthSession newSessionFromTokenExchange(
 
   protected OAuth2Util.AuthSession newSessionFromTokenResponse(
       AuthConfig config, OAuth2Util.AuthSession parent) {
+    RESTClient client = refreshClient;
```

Comment:
nit: seems like an unrelated change

Reply:
Was just trying to avoid 2 volatile reads while 1 is enough. I will revert.